src/main/java/com/bobocode/nasapicturesstealer/service/PictureStealer.java

```diff
@@ -0,0 +1,65 @@
+package com.bobocode.nasapicturesstealer.service;
+
+import com.bobocode.nasapicturesstealer.persistance.entity.Camera;
+import com.bobocode.nasapicturesstealer.persistance.entity.Picture;
+import com.bobocode.nasapicturesstealer.persistance.repository.CameraRepository;
+import com.bobocode.nasapicturesstealer.persistance.repository.PictureRepository;
+import com.bobocode.nasapicturesstealer.service.model.NasaCamera;
+import com.bobocode.nasapicturesstealer.service.model.NasaPicture;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@AllArgsConstructor
+@Service
+public class PictureStealer {
+
+    private final NasaPictureReceiver nasaPictureReceiver;
+    private final CameraRepository cameraRepository;
+    private final PictureRepository pictureRepository;
+
+    @Transactional
```

Comment:
are you specifically doing all the saving in one transaction?  how are you insured against parallel execution?

Reply:
I created 2 services to handle saving for camera and picture with own `@Transactional`